### PR TITLE
Add ESR_EL3 register

### PR DESCRIPTION
**Pull request by rmsyn**

### CODE REVIEW

File: src/registers/esr_el3.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+use tock_registers::{interfaces::Readable, register_bitfields};
+
+register_bitfields! {u64,
+    pub ESR_EL3 [
+
+        /// Reserved
+        RES0 OFFSET(37) NUMBITS(27) [],
+
+        /// Instruction Specific Syndrom 2
```
Comment (nchong-at-aws) [inline src/registers/esr_el3.rs]:
```suggestion
        /// Instruction Specific Syndrome 2
```

File: src/registers/esr_el3.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        /// When FEAT_LS64_V is implemented, if a memory access generated by an ST64BV instruction
+        /// generates a Data Abort exception for a Translation fault, Access flag fault, or Permission fault, then
+        /// this field holds register specifier, Xs.
+        ///
+        /// When FEAT_LS64_ACCDATA is implemented, if a memory access generated by an ST64BV0
+        /// instruction generates a Data Abort exception for a Translation fault, Access flag fault, or Permission
+        /// fault, then this field holds register specifier, Xs.
+        ///
+        /// Otherwise, this field is RES 0.
+        ISS2 OFFSET(36) NUMBITS(5) [],
```
Comment (nchong-at-aws) [inline src/registers/esr_el3.rs]: Offset should be 32

Screenshot 2023-07-18 at 4 13 21 PM

File: src/registers/esr_el3.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        ///   synchronous exceptions.
+        ///
+        ///   - Unused values in the range 0b101101 - 0b111111 (0x2D - 0x3F) are reserved for future use, and
+        ///   might be used for synchronous or asynchronous exceptions.
+        ///
+        /// The effect of programming this field to a reserved value is that behavior is CONSTRAINED UNPREDICTABLE.
+        ///
+        /// The reset behavior of this field is:
+        ///   - On a Warm reset, this field resets to an architecturally UNKNOWN value.
+        EC  OFFSET(26) NUMBITS(6) [
```
Comment (nchong-at-aws) [inline src/registers/esr_el3.rs:57]: .

**Review by nchong-at-aws — CHANGES_REQUESTED** [review]

File: src/registers/esr_el3.rs (hunk 3 above)
Comment (rmsyn) [inline src/registers/esr_el3.rs:57]: ?

**Review by nchong-at-aws — APPROVED** [review]

**Pull request merged**